### PR TITLE
Add instance settings.

### PR DESCRIPTION
**Pull request by cthulahoops**
Addresses: #58

Currently implemented:

* Instance name
* Instance description
* Code of conduct
* Allow registration

I decided to store this in the database so that settings can be easily
changed at runtime through the web interface. (This web interface does
not exist.)

### CODE REVIEW

File: fedireads/migrations/0042_sitesettings.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    operations = [
+        migrations.CreateModel(
+            name='SiteSettings',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('name', models.CharField(default='wyrms.cthulahoops.org', max_length=100)),
+                ('instance_description', models.TextField(default='This instance has no description.')),
+                ('code_of_conduct', models.TextField(default='Add a code of conduct here.')),
+                ('allow_registration', models.BooleanField(default=True)),
```
Comment (mouse-reeve) [inline fedireads/migrations/0042_sitesettings.py:20]: can we add an optional logo field here as well?

Comment (cthulahoops) [reply fedireads/migrations/0042_sitesettings.py:20]: Sure! Storing image data in the database?

I figure there's lots more settings that can be added to this, so this is mainly to get the structure in.

Comment (mouse-reeve) [reply fedireads/migrations/0042_sitesettings.py:20]: I think that works -- thanks so much for building out this feature!

**Comment by mouse-reeve** [comment]
this is reminding me of my vague intention of setting up django admin and superusers. it wouldn't be as nice as a custom interface but it would probably be comparatively quick 🤔

**Comment by cthulahoops** [comment]
> this is reminding me of my vague intention of setting up django admin and superusers. it wouldn't be as nice as a custom interface but it would probably be comparatively quick thinking

Yeah, I wondered about whether to do it that way. It would certainly be quick and easy.

I'm currently working on invites for when registration isn't allowed which might be a bit painful to have to manage via the admin interface. But it'd work.

**Comment by cthulahoops** [comment]
Adding invite code support onto this as it depends on it lots.

File: fedireads/migrations/0042_sitesettings.py (hunk 1 above)
Comment (cthulahoops) [reply fedireads/migrations/0042_sitesettings.py:20]: I might add the logo in separate PR - there's enough in this already.

You're welcome. It's fun to do a feature again after all the hidden auth work!

Comment (mouse-reeve) [reply fedireads/migrations/0042_sitesettings.py:20]: no need to prioritize logo, it just occurred to me

Comment (cthulahoops) [reply fedireads/migrations/0042_sitesettings.py:20]: Cool! Though it might be fun to work out how to store the image data.

**Comment by mouse-reeve** [comment]
sorry for the delay on this, things have been *waves vaguely at current events*

**Comment by cthulahoops** [comment]
No problem. Things very much are.

**Pull request merged**